addons/website/models/mixins.py

```diff
@@ -112,6 +112,34 @@ def _default_cover_properties(self):
             "resize_class": "o_half_screen_height",
         }
 
+    def _get_valid_cover_resize_classes(self):
+        return ['o_half_screen_height', 'o_full_screen_height', 'cover_auto']
+
+    def write(self, vals):
+        if 'cover_properties' not in vals:
+            # Not updating cover properties, normal write.
+            return super().write(vals)
+
+        cover_properties = json_scriptsafe.loads(vals['cover_properties'])
+        resize_class = cover_properties.get('resize_class', '').split()
+        classes = self._get_valid_cover_resize_classes()
+        if any((class_name in resize_class) for class_name in classes):
+            # Updating cover properties and the given resize class is valid,
+            # normal write.
+            return super().write(vals)
+
+        # If we do not receive a valid resize_class via the cover_properties, we
+        # keep the original one (prevents updates on list displays from
+        # destroying resize_class).
+        result = True
+        copy_vals = dict(vals)
+        for item in self:
+            old_cover_properties = json_scriptsafe.loads(item.cover_properties)
+            cover_properties['resize_class'] = old_cover_properties.get('resize_class', classes[0])
+            copy_vals['cover_properties'] = json_scriptsafe.dumps(cover_properties)
+            result &= super(WebsiteCoverPropertiesMixin, item).write(copy_vals)
```

Comment:
Can `write()` return anything else than `True`?
AFAIK, it either goes fine and return `True` or it raises an error. Then it is pointless to store the result of each `write()` in the loop, you can just `return True` at the end as we do in COW (ir.ui.view), no?

Reply:
You are completely correct.  In fact, the return value of `write()` is meaningless.
I think that it dates back when the result had to be serialized by XML-RPC, and the protocol had no way of serializing `None`; so it returned `True`, which can be serialized.  It could have returned `42`, since the return value means nothing.